File: 2021/11/14/DecryptString.java
/**
 * https://leetcode.com/problems/decrypt-string-from-alphabet-to-integer-mapping/
 * @author Andrew Jarombek
 * @since 11/14/2021
 */

public class DecryptString {
    public String freqAlphabets(String s) {
        StringBuilder sb = new StringBuilder();

        int i = 0;
        while (i < s.length()) {
            if (i + 2 < s.length() && s.charAt(i + 2) == '#') {
                sb.append((char) (96 + Integer.parseInt(s.substring(i, i + 2))));
                i += 3;
            } else {
                sb.append((char) (96 + Integer.parseInt(s.substring(i, i + 1))));
                i++;
            }
        }

        return sb.toString();
    }
}

File: 2022/08/05/decrypt_string.py
"""
https://leetcode.com/problems/decrypt-string-from-alphabet-to-integer-mapping/
Author: Andrew Jarombek
Date: 8/5/2022
"""


def freq_alphabets(s: str) -> str:
    chars = []
    index = len(s)

    while index > 0:
        if s[index - 1] == '#':
            chars.append(chr(ord('a') + int(s[index - 3:index - 1]) - 1))
            index -= 3
        else:
            chars.append(chr(ord('a') + int(s[index - 1]) - 1))
            index -= 1

    chars.reverse()
    return ''.join(chars)


if __name__ == '__main__':
    assert freq_alphabets("114#425#") == "andy"
